accounts/abi/bind/bindv2.go

```diff
@@ -0,0 +1,311 @@
+package bind
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"go/format"
+	"regexp"
+	"strings"
+	"text/template"
+	"unicode"
+)
+
+type binder struct {
+	// contracts is the map of each individual contract requested binding
+	contracts map[string]*tmplContractV2
+
+	// structs is the map of all redeclared structs shared by passed contracts.
+	structs map[string]*tmplStruct
+
+	// aliases is a map for renaming instances of named events/functions/errors to specified values
+	aliases map[string]string
+}
+
+// registerIdentifier applies alias renaming, name normalization (conversion to camel case), and registers the normalized
+// name in the specified identifier map.  It returns an error if the normalized name already exists in the map.
+func (b *contractBinder) registerIdentifier(identifiers map[string]bool, original string) (normalized string, err error) {
+	normalized = methodNormalizer(alias(b.binder.aliases, original))
+	// Name shouldn't start with a digit. It will make the generated code invalid.
+	if len(normalized) > 0 && unicode.IsDigit(rune(normalized[0])) {
+		normalized = fmt.Sprintf("E%s", normalized)
+		normalized = abi.ResolveNameConflict(normalized, func(name string) bool {
+			_, ok := identifiers[name]
+			return ok
+		})
+	}
+
+	if _, ok := identifiers[normalized]; ok {
+		return "", fmt.Errorf("duplicate symbol '%s'", normalized)
+	}
+	identifiers[normalized] = true
+	return normalized, nil
+}
+
+// RegisterCallIdentifier applies registerIdentifier for contract methods.
+func (b *contractBinder) RegisterCallIdentifier(id string) (string, error) {
+	return b.registerIdentifier(b.callIdentifiers, id)
+}
```

Comment:
WHy is this public accessor needed? Are users expected to use this? In what scenario?

Reply:
it's a public method on a private type.  Will move it to lowercase.

Reply:
> it's a public method on a private type. Will move it to lowercase.

i.e.  I didn't mean it for public consumption.  a mistake on my part.

Reply:
Well, if it's not for public, is it even needed? doesn't save a whole lot of typing really

Reply:
not really needed no.